equistore-core/src/labels.rs

```diff
@@ -355,4 +466,82 @@ mod tests {
         let e = LabelsBuilder::new(vec!["not", "there", "not"]).err().unwrap();
         assert_eq!(e.to_string(), "invalid parameter: labels names must be unique, got 'not' multiple times");
     }
+
+    #[test]
+    fn union() {
```

Comment:
Are you handling the case when one of the labels is an empty label?

Reply:
added a test!